diff --git a/googletest/cmake/internal_utils.cmake b/googletest/cmake/internal_utils.cmake
index 41405587..637a3d67 100644
--- a/googletest/cmake/internal_utils.cmake
+++ b/googletest/cmake/internal_utils.cmake
@@ -258,11 +258,21 @@ endif()
 
 # Sets PYTHONINTERP_FOUND and PYTHON_EXECUTABLE.
 if ("${CMAKE_VERSION}" VERSION_LESS "3.12.0")
+  message(FATAL_ERROR "Should not pass here")
   find_package(PythonInterp)
 else()
-  find_package(Python COMPONENTS Interpreter)
-  set(PYTHONINTERP_FOUND ${Python_Interpreter_FOUND})
-  set(PYTHON_EXECUTABLE ${Python_EXECUTABLE})
+  if(NOT Python_FOUND AND NOT Python3_FOUND)
+    message(FATAL_ERROR "Should not pass here")
+    find_package(Python COMPONENTS Interpreter)
+  endif()
+
+  if(Python_FOUND)
+    set(PYTHONINTERP_FOUND ${Python_Interpreter_FOUND} CACHE INTERNAL "" FORCE)
+    set(PYTHON_EXECUTABLE ${Python_EXECUTABLE} CACHE INTERNAL "" FORCE)
+  elseif(Python3_FOUND)
+    set(PYTHONINTERP_FOUND ${Python3_Interpreter_FOUND} CACHE INTERNAL "" FORCE)
+    set(PYTHON_EXECUTABLE ${Python3_EXECUTABLE} CACHE INTERNAL "" FORCE)
+  endif()
 endif()
 
 # cxx_test_with_flags(name cxx_flags libs srcs...)
